config/config-api/src/com/thoughtworks/go/config/AgentConfig.java

```diff
@@ -165,6 +170,10 @@ public boolean isNull() {
         return false;
     }
 
+    public AgentConfig deepClone() {
+        return new Cloner().deepClone(this);
```

Comment:
I see that this method is used in method ```findPendingAgents```. Why does the agent config need to be cloned to add to the pending agent list?